action.py

```diff
@@ -212,6 +220,18 @@ def create_beanstalk_application_version(app: str, version: str, description: st
         Process=True,
     )
 
+    steps = int(math.ceil(timeout / wait_time))
+    for step in range(steps):
```

Comment:
Rename `timeout` to `max_retries` then use
```suggestion
retries = 0
while retries < max_retries:
    ...
    retries += 1
```